File: frontend/src/lib/pipeline.ts
import { Parameter } from './parameter';

export interface Pipeline {
  createAt?: string;
  id?: number;
  author: string;
  description: string;
  ends: number;
  name: string;
  packageId: number;
  parameters: Parameter[];
  recurring: boolean;
  recurringIntervalHours: number;
  starts: number;
}

File: apiserver/src/message/pipelinemanager/pipeline.go
package pipelinemanager

type Pipeline struct {
	*Metadata               `json:",omitempty"`
	Name        string      `json:"name" gorm:"not null"`
	Description string      `json:"description,omitempty"`
	PackageId   uint        `json:"packageId" gorm:"not null"`
	Parameters  []Parameter `json:"parameters,omitempty" gorm:"polymorphic:Owner;"`
}
